src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -1721,14 +1720,28 @@ bool canProceed()
     return proceed;
 }
 
+static int isSudo()
+{
+    const char* SudoMessage = "ERROR: root privileges required.";
+    if ((getuid() == 0) || (geteuid() == 0))
+        return 0;
+    std::cout << SudoMessage << std::endl;
+    return -EPERM;
+}
+
 int xcldev::xclReset(int argc, char *argv[])
 {
     int c;
     unsigned index = 0;
+    bool all = false;
     const std::string usage("Options: [-d index]");
 
-    while ((c = getopt(argc, argv, "d:")) != -1) {
+    while ((c = getopt(argc, argv, "ad:")) != -1) {
```

Comment:
all/'a' looks like a new feature. Do we really need this?

Reply:
this is a hidden option that is used to distinguish whether the 'xbutil reset' is from user or from 'xbreset' since we have path 'xbutil reset' -> xbreset -> 'xbutil reset'